dragonchain/webserver/lib/dragonnet.py

```diff
@@ -87,6 +89,10 @@ def process_receipt_v1(block_dto: Dict[str, Any]) -> None:
                     matchmaking.add_receipt(l1_block_id, level_received_from, dc_id, block_id, proof)
                 except Exception:
                     _log.exception("matchmaking add_receipt failed!")
+                if level_received_from == 5:
+                    client = redisearch._get_redisearch_index_client(redisearch.Indexes.verification.value)
+                    client.redis.sadd(redisearch.L5_NODES, block_model.dc_id)
+                    block_dao.insert_l5_verification(storage_location, block_model)
```

Comment:
This should definitely only happen after the broadcast system processes this block, since the received verification in the broadcast system can be rejected, in which case you definitely don't want to save this.

Reply:
great point